src/dotnet/APIView/APIViewWeb/wwwroot/js/site.js

```diff
@@ -25,18 +25,61 @@ $(function () {
                 thread.after(commentForm);
             }
             else {
-                commentForm.insertAfter(thisRow).wrap("<tr>").wrap("<td>");
+                commentForm.insertAfter(thisRow).wrap("<tr>").wrap("<td colspan=\"2\">");
             }
         }
 
         commentForm.show();
         commentForm.find(".id-box").val(id);
         commentForm.find(".new-thread-comment-text").focus();
         commentForm.find(".comment-cancel-button").click(function () { hideCommentBox(id); });
+        commentForm.find(".comment-submit-button").click(function () {
+            $.ajax({
+                type: "POST",
+                data: commentForm.find("form").serialize()
+            }).done(function (partialViewResult) {
+                updateCommentThread(thisRow.next(), partialViewResult);
+            });
+            return false;
+        });
 
         nextRow.find(".review-thread-reply").hide();
     }
 
+    function updateCommentThread(commentBox, partialViewResult) {
+        partialViewResult = $.parseHTML(partialViewResult);
+        $(commentBox).replaceWith(partialViewResult);
+        $(partialViewResult).find(".review-thread-reply-button").click(function () {
+            showCommentBox($(this).data("element-id"));
+        });
+        $(partialViewResult).find(".comment-delete-button-enabled").click(function () {
+            deleteComment(this.id);
+            return false;
+        });
+    }
+
+    function deleteComment(id) {
+        let button = document.getElementById(id);
+        let commentBox = $(button).parents(".comment-box").first();
+        let gotResponse = false;
+        $.ajax({
+            type: "POST",
+            url: "?handler=delete",
+            data: $(button).parents("form").serialize()
+        }).done(function (partialViewResult) {
+            gotResponse = true;
+            updateCommentThread(commentBox, partialViewResult);
+        });
+        if (!gotResponse) {
```

Comment:
Why?

Reply:
This was an attempt at checking if a partial view result was actually returned by the deletion handler. The idea was to only return a non-null response if there were comments remaining in that element's comment thread - otherwise, I would remove the `comment-box` element entirely. I didn't realize when I pushed this that the `.done()` segment wasn't being skipped because of a null response, but was actually just running asynchronously. So, in short, this fix didn't work. My most recent push has an alternate (working) approach that just checks for any `review-comment` elements in the `comment-box` returned in the `partialViewResult` and removes the thread if none exist.